src/analyzer.rs

```diff
@@ -394,8 +403,19 @@ fn array_recurse_for_type(
                 let inner_array_type = s.type_.clone().unwrap_or_default();
                 return match inner_array_type.as_ref() {
                     "object" => {
-                        let structsuffix = uppercase_first_letter(key);
-                        Ok((format!("Vec<{}{}>", stack, structsuffix), level))
+                        let mut dict_value = None;
+                        if let Some(additional) = &s.additional_properties {
+                            dict_value = resolve_additional_properties(additional, stack, key, s)?;
+                        }
+
+                        let vec_value = if let Some(dict_value) = dict_value {
+                            format!("BTreeMap<String, {dict_value}>")
+                        } else {
+                            let structsuffix = uppercase_first_letter(key);
+                            format!("{stack}{structsuffix}")
+                        };
```

Comment:
the continuation here is quite a bit different from the first invocation of `resolve_additional_properties` but that should help us in the future 👍

Reply:
Yes, unfortunately. And I am not sure which edge cases are still affected here, that's why I simply ignored it and we should go and fix it if it comes up.

I'll add a comment to clarify this

Reply:
Actually I think this handles all cases equally now, the `x_kubernetes_preserve_unknown_fields` special case is handled above.